.github/workflows/build_cli_image_on_push.yml

```diff
@@ -0,0 +1,30 @@
+name: Build and publish cli image on push
+
+on: [push]
+
+jobs:
+  docker-image-push:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out git repository
+        uses: actions/checkout@v3
+
+      - name: Extract branch name and remove illegal chars
+        id: get_branch_name
+        shell: bash
+        run: echo "branch=$(tr "/" "-" <<<${GITHUB_REF#refs/heads/})" >> $GITHUB_ENV
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Build and push
+        id: docker_cli_build
+        uses: docker/build-push-action@v3
+        with:
+          context: ./
+          file: ./Dockerfile.cli
+          push: true
+          tags: "clinicalgenomics/cg-cli:${{ env.branch }}"
```

Comment:
If we want to follow the same way we do with the other dockerfile we would have a cg-stage-cli repository which we publish to on push and then a cg-cli repository which is published to on release. But maybe we can push to the same ones we have but have `cli` as a tag? What do you think is best?

Reply:
The only difference between the stage and prod cli is the config file we point to at runtime. So I don't think it makes sense to have separate stage/prod images. I think it seems intuitive to keep the web api and cli images in separate repositories though, what do you think? It could also make sense to keep all images in one repository and tag them clearly.

Currently, I think the best approach is to use a single new repository called `cg-cli` like

Feature branches: clinicalgenomics/cg-cli:<branch-name>
Master branch: clinicalgenomics/cg-cli:<version> and clinicalgenomics/cg-cli:latest

Reply:
Sure I like that suggestion of a CLI repo, I don't think tags are supposed to be used the way I thought. But he same does hold true for the API though (ie no need for separate stage prod repos). Maybe we could merge those as well, while we're at it? @henrikstranneheim do you have any thoughts on this?

Reply:
Yes, that is how tagging is used here as well! So the conclusion is that we want to merge the `clinicalgenomics/cg` and `clinicalgenomics/cg-stage`? Does it affect this PR? I think we have to separate the images for the cg web api and cli images though since the Dockerfiles are different

Reply:
I agree with you @seallard. I think we should have two separate repos for CLI and API

Reply:
Alright, I'm flip flopping between the options haha! I'll go for the separate `cg-cli` repo for now and we can adjust it in the future.